src/api/services/participantsService.ts

```diff
@@ -75,30 +75,48 @@ export const getParticipantsBySiteIds = async (siteIds: number[]) => {
 
 export const addSharingParticipants = async (
   participantSiteId: number,
-  siteIds: number[],
-  types: ClientType[]
+  siteIds: number[]
 ): Promise<SharingListResponse> => {
   const sharingListResponse = await getSharingList(participantSiteId);
-  const newSharingSet = new Set([...sharingListResponse.allowed_sites, ...siteIds]);
+  const existingSharedSites = sharingListResponse.allowed_sites ?? [];
+  const newSharingSet = new Set([...existingSharedSites, ...siteIds]);
   const response = await updateSharingList(
     participantSiteId,
     sharingListResponse.hash,
-    [...newSharingSet],
-    types
+    Array.from(newSharingSet),
+    sharingListResponse.allowed_types
   );
   return response;
 };
 
 export const deleteSharingParticipants = async (
   participantSiteId: number,
-  siteIds: number[],
-  types: ClientType[]
+  siteIds: number[]
 ): Promise<SharingListResponse> => {
   const sharingListResponse = await getSharingList(participantSiteId);
+  if (!sharingListResponse.allowed_sites) throw Error('The site is not shared with any sites');
```

Comment:
I'm wondering if it would be clearer if the error message was something like:
"There are no sharing sites to delete"

Reply:
yep I agree, I think this error message is more clear to me